borg/hashindex.pyx

```diff
@@ -269,16 +269,11 @@ cdef class ChunkIndex(IndexBase):
         self._add(<char*> key, data)
 
     cdef _add(self, void *key, uint32_t *data):
-        cdef uint32_t refcount1, refcount2, result
+        cdef uint64_t result64
         values = <uint32_t*> hashindex_get(self.index, key)
         if values:
-            refcount1 = _le32toh(values[0])
-            refcount2 = _le32toh(data[0])
-            if _MAX_VALUE - refcount1 < refcount2 or _MAX_VALUE - refcount2 < refcount1:
-                result = _MAX_VALUE
-            else:
-                result = refcount1 + refcount2
-            values[0] = _htole32(result)
+            result64 = <uint64_t> _le32toh(values[0]) + _le32toh(data[0])
+            values[0] = _htole32(min(result64, _MAX_VALUE))
```

Comment:
This way it's much clearer as @verygreen suggested. Note that if (for some reason) we actually make use of one of the ~1000 reserved values, then this needs changing. But since we don't know what semantics these "maybe" future values have this is okay by me.